java/fury-core/src/main/java/org/apache/fury/serializer/ObjectSerializer.java

```diff
@@ -495,104 +548,165 @@ static boolean writePrimitiveFieldValueFailed(
    * @return true if field value isn't written by this function.
    */
   static boolean writeBasicObjectFieldValueFailed(
-      Fury fury, MemoryBuffer buffer, Object fieldValue, short classId) {
+      Fury fury,
+      MemoryBuffer buffer,
+      Object fieldValue,
+      short classId,
+      FuryFieldAnnotationInfo fieldAnnotationInfo) {
     if (!fury.isBasicTypesRefIgnored()) {
       return true; // let common path handle this.
     }
+    boolean nullable = fieldAnnotationInfo.nullable;
     // add time types serialization here.
     switch (classId) {
-      case ClassResolver.STRING_CLASS_ID: // fastpath for string.
-        fury.writeJavaStringRef(buffer, (String) (fieldValue));
+      case ClassResolver.STRING_CLASS_ID: // fast path for string.
+        fury.writeNullableJavaStringRef(buffer, (String) (fieldValue), nullable);
```

Comment:
This makes writeNullableJavaStringRef too complicated. How about:

```java 
{
String v =  (String) (fieldValue);
if (nullable) {
  fury.writeJavaStringRef(buffer, v);
} else {
  fury.writeString(buffer, v);
}
return false;
}
```

Reply:
Call fury.writeString(buffer, v) separately; This method will ignore the case that stringSerializer.needToWriteRef() is true, resulting in this refResolver.writeRefOrNull(buffer, str) method not being called. Is this as expected?

Reply:
How about
```java
if (nullable || trackStringRef) {
  fury.writeJavaStringRef(buffer, v);
} else {
  fury.writeString(buffer, v);
}
```